cromulent/model.py

```diff
@@ -745,9 +759,10 @@ def _toJSON(self, done, top=None):
 			del d['context']
 
 		# Check mandatory properties
-		for e in self._required_properties:
-			if e not in d:
-				raise RequirementError("Resource type '%s' requires '%s' to be set" % (self._type, e), self)
+		if self._factory.validate_profile:
+			for e in self._required_properties:
```

Comment:
What is `_requried_properties`? It doesn't seem to ever have values added to it(?).

Reply:
It's there in case we want to require that every instance of a class always have a particular property.  I started by just deleting this all completely as we don't do that ... but the cost to evaluate either a flag on factory or iterate an empty list seems minimal.